client/client-multi/client-c/src/kaa/kaa_user.c

```diff
@@ -426,6 +605,112 @@ kaa_error_t kaa_user_handle_server_sync(kaa_user_manager_t *self
                     (self->attachment_listeners.on_detached)(self->attachment_listeners.context, access_token);
                 break;
             }
+            case ENDPOINT_ATTACH_RESPONSES_FIELD: {
+                uint16_t attach_responses_count = (field_header) & 0xFFFF;
+
+                uint8_t  result_code;
+                uint8_t  options;
+                uint16_t request_id;
+                char     endpoint_id[KAA_ENDPOINT_ID_LENGTH+1];
+
+                if (sizeof(uint32_t) > remaining_length)
+                    return KAA_ERR_INVALID_BUFFER_SIZE;
+
+                for (uint32_t i = 0; i < attach_responses_count; ++i) {
+                    result_code = *(reader->current++);
+                    options     = *(reader->current++);
+                    request_id  = KAA_NTOHS(*(uint16_t*)reader->current);
+                    reader->current += sizeof(uint16_t);
+                    remaining_length -= sizeof(uint32_t);
+
+                    if (result_code == USER_RESULT_FAILURE) {
+                        kaa_list_node_t *node = kaa_list_begin(self->attach_endpoints);
+
+                        while (node) {
+                            endpoint_info_t *info = (endpoint_info_t*)kaa_list_get_data(node);
+                            if (info->request_id == request_id) {
+                                kaa_list_remove_at(self->attach_endpoints, node, NULL);
```

Comment:
There is a memory leak. Need to deallocate memory used by the endpoint_token field of an endpoint_info_t structure.